renderer/BufferObject.cpp

```diff
@@ -0,0 +1,221 @@
+/*****************************************************************************
+The Dark Mod GPL Source Code
+
+This file is part of the The Dark Mod Source Code, originally based
+on the Doom 3 GPL Source Code as published in 2011.
+
+The Dark Mod Source Code is free software: you can redistribute it
+and/or modify it under the terms of the GNU General Public License as
+published by the Free Software Foundation, either version 3 of the License,
+or (at your option) any later version. For details, see LICENSE.TXT.
+
+Project: The Dark Mod (http://www.thedarkmod.com/)
+
+$Revision$ (Revision of last commit)
+$Date$ (Date of last commit)
+$Author$ (Author of last commit)
+
+******************************************************************************/
+
+#include "precompiled.h"
+#pragma hdrstop
+#include "tr_local.h"
+#include "BufferObject.h"
+
+/*
+================================================================================================
+
+Buffer Objects
+
+================================================================================================
+*/
+
+/*
+========================
+UnbindBufferObjects
+========================
+*/
+void UnbindBufferObjects() {
+	qglBindBufferARB( GL_ARRAY_BUFFER_ARB, 0 );
+	qglBindBufferARB( GL_ELEMENT_ARRAY_BUFFER_ARB, 0 );
+}
+
+/*
+========================
+CopyBuffer
+========================
+*/
+#ifdef ID_WIN_X86_SSE2_INTRIN
```

Comment:
We use __SSE2__ macro for checking for SSE2 compile-time support.
It is builtin on GCC, and I have added support for it on MSVC.
Could you please replace the macro?

Reply:
Absolutely. Or maybe I'll even just remove that part of code. Haven't been able to see any advantages of it in profiling...

Reply:
I think it 1) copies a bit faster 2) makes sure that write destination does NOT get into CPU cache.
Not sure if the second point makes code faster. I recall that some special types of memory (like "pinned" or something like that) are better accessed with non-caching writes.

Reply:
I changed the flag.